### PR TITLE
FIx pre-commit failing - Add `B023` to `.flake8` ignores

### LINKED ISSUE #7940: Pre-commit failing on genetic_algoirthm/basic_string
### Repository commit

f512b4d105b6f3188deced19761b6ed288378f0d

### Python version (python --version)

Python 3.10.8

### Dependencies version (pip freeze)

.

### Expected behavior

`return (item, float(score))  #  noqa: B023`

### Actual behavior

`return (item, float(score))  #  noqa: B023` is stripped by 
```
 repo: https://github.com/asottile/yesqa
    rev: v1.4.0
    hooks:
      - id: yesqa
        additional_dependencies:
          *flake8-plugins
```

### PR DESCRIPTION
**Pull request by CaedenPH**
### Describe your change:
Fixes #7940

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: genetic_algorithm/basic_string.py [hunk 1]
```diff
@@ -129,9 +129,7 @@ def select(parent_1: tuple[str, float]) -> list[str]:
             child_n = int(parent_1[1] * 100) + 1
             child_n = 10 if child_n >= 10 else child_n
             for _ in range(child_n):
-                parent_2 = population_score[  # noqa: B023
-                    random.randint(0, N_SELECTED)
-                ][0]
+                parent_2 = population_score[random.randint(0, N_SELECTED)][0]
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:132]:
```suggestion
                random_x = random.randint(0, N_SELECTED)
                parent_2 = population_score[random_x][0]
```

File: .flake8 [hunk 2]
```diff
@@ -6,3 +6,4 @@ extend-ignore =
     # Formatting style for `black`
     E203  # Whitespace before ':'
     W503  # Line break occurred before a binary operator
+    B023  # Function definition does not bind loop variable
```
Comment (cclauss) [inline .flake8:9]: The ⛔ is because GitHub expects text files to end in one and only one `\n`.

Comment (CaedenPH) [reply .flake8:9]: I don't understand

Comment (cclauss) [reply .flake8:9]: See the ⛔ here.

Comment (cclauss) [inline .flake8:9]:
```suggestion
```

**Comment by CaedenPH** [comment]
#@cclauss the point is that those code snippets raise `B023`, which is automatically stripped by yesqa for some reason, failing the pre commit

**Comment by CaedenPH** [comment]
Huh, seems you can't do in-line comments in a `.flake8` file

**Comment by CaedenPH** [comment]
@cclauss Lots of failures in pre-commit due to previous code.
I wonder why they're only coming up now
Should I fix these issues?

**Comment by cclauss** [comment]
I think you should close this PR and start from scratch.

I am worried about ignoring all instances of `B023` because [the problem is real](https://docs.python-guide.org/writing/gotchas/#late-binding-closures) even though [there are some false alarms](https://github.com/PyCQA/flake8-bugbear/pull/303).  I would be happier to see us using `# noqa: B023` in specific situations rather than a project-wide ignore.

**Comment by CaedenPH** [comment]
> 

Starting again sounds like a good idea but I'm not entirely sure how to fix this without ignoring all instances and trusting that the doctests will keep the algorithms on the straight and narrow.
It is another gotcha within our pre-commit that when you add `#noqa: B023` to the actual error that the `yesqa` strips it...

**Comment by cclauss** [comment]
# yesqa might be stripping because PyCQA/flake8-bugbear#303 was recently landed and it might have fixed false alarms.

**Pull request closed without merging**